src/scatter_buildblock/ScatterEstimation.cxx

```diff
@@ -0,0 +1,1486 @@
+            dynamic_cast<const ChainedBinNormalisation*>(this->multiplicative_binnorm_sptr.get());
+
+    if (!is_null_ptr(tmp_chain_multiplicative_binnorm_sptr ))
+    {
+        do_chains = true;
+        tmp_atten_projdata_sptr =
+                dynamic_cast<BinNormalisationFromProjData*> (tmp_chain_multiplicative_binnorm_sptr->get_second_norm().get())->get_norm_proj_data_sptr();
+    }
+    else
+    {
+        do_chains = false;
+        tmp_atten_projdata_sptr =
+                dynamic_cast<BinNormalisationFromProjData*> (this->multiplicative_binnorm_sptr.get())->get_norm_proj_data_sptr();
+    }
+
+#if 0
+    info("ScatterEstimation: 3.Calculating the attenuation projection data...");
+
+    if( tmp_atten_projdata_sptr->get_num_segments() > 1)
+    {
+        info("ScatterEstimation: Running SSRB on attenuation correction coefficients ...");
+
+        std::string out_filename = extras_path.get_path() + "tmp_atten_sino_2d.hs";
+
+        atten_projdata_2d_sptr.reset(new ProjDataInterfile(this->input_projdata_2d_sptr->get_exam_info_sptr(),
+                                                           this->input_projdata_2d_sptr->get_proj_data_info_sptr()->create_shared_clone(),
+                                                           out_filename,
+                                                           std::ios::in | std::ios::out | std::ios::trunc));
+        SSRB(*atten_projdata_2d_sptr,
+             *tmp_atten_projdata_sptr, true);
+    }
+    else
+    {
+        // TODO: this needs more work. -- Setting directly 2D proj_data is buggy right now.
+        atten_projdata_2d_sptr = tmp_atten_projdata_sptr;
+    }
+#else
+    {
+        std::string in_filename = extras_path.get_path() + "tmp_atten_sino_2d.hs";
+        atten_projdata_2d_sptr = ProjData::read_from_file(in_filename);
+    }
+#endif
+
+    if(!do_chains)
+    {
+        multiplicative_binnorm_2d_sptr.reset(new BinNormalisationFromProjData(atten_projdata_2d_sptr));
+        multiplicative_binnorm_2d_sptr->set_up(this->input_projdata_2d_sptr->get_proj_data_info_sptr()->create_shared_clone());
+    }
+#if 0
+    else
+    {
+        if(!tmp_chain_multiplicative_binnorm_sptr->is_first_trivial()) // Check that we have actually something in here
+        {
+            shared_ptr<ProjData> tmp_norm_projdata_sptr;
+            // Check if 3D
+            const BinNormalisationFromProjData* tmp  =
+                    dynamic_cast<BinNormalisationFromProjData*> (tmp_chain_multiplicative_binnorm_sptr->get_first_norm().get());
+
+
+            if(!is_null_ptr(tmp))
+            {
+                tmp_norm_projdata_sptr = tmp->get_norm_proj_data_sptr();
+            }
+            else
+            {
+                error("ScatterEstimation: ECAT8 Normsalisation factors from are supported, but we cannot SSRB directly.\n"
+                      "Please convert to ProjData before proceeding or use 3D estimation. Aborting.");
+
+            }
```

Comment:
This is too restrictive. BinNormalisationFromECAT8 and the others will actually do the proper thing on SSRB data (i.e. the equivalent of the calculation below). So, you can just set `norm_coeff_2d_sptr` to `tmp->get_norm_proj_data_sptr()` (or to be safe, a `clone` of that one, but I guess we cannot do that yet)

Reply:
I'm in any case confused about the norm situation. I'll try to summarise this in another comment

Reply:
I am not sure if this comment is the same as the one above. 
Indeed if you have pre-SSRB your ecat8 norm factors you are good to go. 
But if not, you cannot SSRB the ecat8 directly. 
Thats why if we need to SSRB I ask for ProjData.